### DISCUSSION TITLE
balance/balance-topology

### DISCUSSION THREAD

**Discussion by giscus** [isno/theByteBook#218] [General]
# balance/balance-topology

构建大规模高可用的分布式系统

https://www.thebyte.com.cn/balance/balance-topology.html

**Comment by fFee-ops** [comment]
请教下：“这种模式下负载均衡器（即使后端已经做了集群）是单点的” ，假设我使用主备模式、分布式部署这种方案，它还是有单点故障码？为什么？

**Reply by linjiaqin12138** [reply]
我觉得接收流量的入口怎么样都是单点的，再怎么分布式，最终都是一个统一的入口接收。除非DNS解析到不同的入口上。

**Reply by joy717** [reply]
LB本身是单点的，就会有单点故障的问题. 毕竟所有的流量都要经过LB，那么LB挂了，就无法将请求发往LB的后端了
但是也可以针对LB做高可用，比如用keepalived做 LB的主备.  这个就可以避免单点故障.

**Comment by xuxu97** [comment]
6.3.2 边缘代理型
——————————
4.3.2

**Comment by CodeMonster-lil** [comment]
边车代理型图4-8，Sidecar Proxy
